File: ai/app/common/api_response.py
from typing import Optional, TypeVar, Generic
from pydantic import BaseModel

T = TypeVar('T')

class ApiResponse(BaseModel, Generic[T]):
    status: str
    message: str
    data: Optional[T] = None

    @staticmethod
    def success(message: str = "요청이 성공적으로 처리되었습니다", data: Optional[T] = None):
        return ApiResponse(status="SUCCESS", message=message, data=data)

    @staticmethod
    def fail(message: str = "요청 처리 중 오류가 발생했습니다", data: Optional[T] = None):
        return ApiResponse(status="ERROR", message=message, data=data)

File: backend/src/main/java/com/ssafy/a202/common/entity/ApiResponse.java
package com.ssafy.a202.common.entity;

import com.ssafy.a202.global.constants.ErrorCode;
import com.ssafy.a202.global.constants.Status;
import com.ssafy.a202.global.constants.SuccessCode;

public record ApiResponse<T>(
        Status status,
        String message,
        T data
) {

    public static <T> ApiResponse<T> success(SuccessCode successCode, T data) {
        return new ApiResponse<>(Status.SUCCESS, successCode.getMessage(), data);
    }

    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(Status.SUCCESS, SuccessCode.SUCCESS_DEFAULT.getMessage(), data);
    }

    public static ApiResponse<Void> success(SuccessCode successCode) {
        return new ApiResponse<>(Status.SUCCESS, successCode.getMessage(), null);
    }

    public static ApiResponse<Void> success() {
        return new ApiResponse<>(Status.SUCCESS, SuccessCode.SUCCESS_DEFAULT.getMessage(), null);
    }

    public static <T> ApiResponse<T> fail(ErrorCode errorCode, T data) {
        return new ApiResponse<>(Status.ERROR, errorCode.getMessage(), data);
    }

    public static ApiResponse<Void> fail(ErrorCode errorCode) {
        return new ApiResponse<>(Status.ERROR, errorCode.getMessage(), null);
    }
}